src/enclave/enclave_oe.c

```diff
@@ -253,38 +251,63 @@ static void _read_eeid_config()
     sgxlkl_enclave_state.config = cfg;
 }
 
+#define CHECK_INSIDE(X, S)                                  \
```

Comment:
Maybe move these macros into `enclave_util.h`?  These don't appear to include anything from the surrounding environment and so should be `static inline` functions, not macros.

Please also add a message argument and use `sgxlkl_fail` to print a useful error (or, better, stringify the arguments and use those in the error message - in which case it would have to remain a macro).